WalletWasabi.Tests/Helpers/WabiSabiFactory.cs

```diff
@@ -33,7 +33,9 @@ public static Coin CreateCoin(Key? key = null, Money? amount = null, ScriptPubKe
 	{
 		key ??= new();
 		amount ??= Money.Coins(1);
-		return new(new OutPoint(Hashes.DoubleSHA256(key.PubKey.ToBytes()), 0), new TxOut(amount, key.PubKey.GetScriptPubKey(scriptPubKeyType)));
+		return new(
+			new OutPoint(Hashes.DoubleSHA256(key.PubKey.ToBytes().Concat(BitConverter.GetBytes(amount)).ToArray() ), 0),
+			new TxOut(amount, key.PubKey.GetScriptPubKey(scriptPubKeyType)));
```

Comment:
Can you elaborate why this change was necessary?

I just want to understand it.

Reply:
Oh, I see now. I missed out that this is in `Tests/Helpers/`.

Thanks Kimi for the explanation.

Reply:
@kiminuo explained this perfectly well. The only thing to add is that _before_ the **exact same coin** added in two different moments were considered as two completely different coins. _After_ this change they are considered equal and then like the same coin.